include/envoy/server/config_tracker.h

```diff
@@ -53,6 +54,27 @@ class ConfigTracker {
    * @return EntryOwnerPtr the new entry's owner object. nullptr if the key is already present.
    */
   virtual EntryOwnerPtr add(const std::string& key, Cb cb) PURE;
+
+  /**
+   * Add or update a managed config to the config tracker under the given key
+   * @param key the map key for the configuration.
+   * @param message the message to be managed by config tracker.
+   */
+  virtual void addOrUpdateManagedConfig(const std::string& key,
```

Comment:
This seems like maybe too general an interface; rather than a generic key-value map, you are modeling a service endpoint : service config relationship, so it would make sense to use concrete types to benefit from type safety.

Reply:
We could do that - but I thought this generic interface would help to add any more things like this in future because `ConfigTracker` can track any such configs. But if you are worried that it is too generic, we can change to a specific thing - Let me know.

Reply:
There is a risk of being overly generic for sure and descending into Lisp :) I think we have a concrete need here for map from service endpoint to the concrete control plane info proto, so that would be preferable.

Reply:
No issues. Will change it.